plugin/trino-hudi/src/test/java/io/trino/plugin/hudi/TestHudiSmokeTest.java

```diff
@@ -476,6 +479,35 @@ public void testPartitionFilterRequiredWithLike()
                 "Filter required on tests." + HUDI_COW_PT_TBL.getTableName() + " for at least one of the partition columns: dt, hh");
     }
 
+    @Test
+    public void testDynamicFilterPredicatePushdown()
+    {
+        final String tableIdentifier = "hudi:tests.hudi_multi_fg_pt_mor";
+        Session session = withMdtDisabled(getSession());
+        MaterializedResult explainRes = getQueryRunner().execute(session,
+                "EXPLAIN ANALYZE SELECT t1.* FROM "
+                        + HUDI_MULTI_FG_PT_MOR + " t1 " +
+                        "INNER JOIN " + HUDI_MULTI_FG_PT_MOR + " t2 ON t1.id = t2.id " +
+                        "WHERE t2.price <= 102");
```

Comment:
Should we add a test to exercise this query involving applying the dynamic filtering?

Reply:
Yes, to make sure end-to-end the queries work with dynamic filtering.